File: src/1_2.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
int main(){
    char *line = NULL;
    size_t len = 0;
    ssize_t read;
    int max[] = { 0, 0, 0 };
    int cur = 0;
    while((read = getline(&line, &len, stdin)) != -1){
        if(!strcmp(line, "\n")){
            for(int i = 0; i < 3; i++){
                if(cur > max[i]){
                    for(int k = 2 - i; k > 0; k--){
                        max[k] = max[k - 1];
                    }
                    max[i] = cur;
                    break;
                }
            }
            cur = 0;
            continue;
        }
        cur += atoi(line);
    }
    printf("%d\n", max[0] + max[1] + max[2]);
    return 0;
}

File: src/1_2.rs
use std::io;
fn main(){
    let lines = io::stdin().lines();
    let mut max: [i32; 3] = [0; 3];
    let mut cur = 0;
    for line in lines {
        if line.as_ref().unwrap().is_empty(){
            for i in 0..3{
                if cur > max[i]{
                    for k in (1..3 - i).rev(){ // terrible for reading! i just want this: for(int k = 2 - i; k > 0; k--)
                        max[k] = max[k - 1];
                    }
                    max[i] = cur;
                    break;
                }
            } 
            cur = 0;
            continue;
        }
        cur += line.unwrap().parse::<i32>().unwrap();
    }
    println!("{}", max.iter().sum::<i32>());
}
